### PR TITLE
feat: Use Next.js in studio-next

### LINKED ISSUE #684: Minimal studio App with a working editor and AsyncAPI documentation
As part of the discussion in #661, I'm creating this issue to track the migration since it's a huge task.

Related to #661

### PR DESCRIPTION
**Pull request by KhudaDad414**
**Description**
- this PR ports the studio from CRA to Next.js.
- it should compile and run in the dev environment.
- It should build without issues.

Some known issues and possible improvements:

- [ ] docker container file is missing in studio-next.
- [ ] tests are missing in studio-next after NEXT.JS migration.
- [ ] Code generation doesn't work in studio-next.
- [ ] Use CodeMirror instead of Monaco.

fixes #684

### CODE REVIEW

**Comment by smoya** [comment]
@KhudaDad414 Besides this PR, is anything else required to have Studio working on Next-JS ? I mean, If we merge this "today", will everything work out of the box?

**Comment by KhudaDad414** [comment]
@smoya Yes, it should work as expected.

**Comment by smoya** [comment]
> @smoya Yes, it should work as expected.

I hope this gets merged soon 🤞

File: apps/studio-next/package.json [hunk 1]
```diff
@@ -9,25 +9,80 @@
     "lint": "next lint"
   },
   "dependencies": {
-    "@codemirror/state": "^6.4.1",
```
Comment (Amzani) [inline apps/studio-next/package.json:12]: @KhudaDad414 why codemirror is revered. It was added by @Shurtu-gal in https://github.com/asyncapi/studio/pull/997

Comment (KhudaDad414) [reply apps/studio-next/package.json:12]: I don't think it's a great idea to push for the CodeMirror at this stage. Migrating to CodeMirror would make the task much much more challenging and this task is already huge.

The CodeMirror was added as a separate code editor and it wasn't configured with the codebase in any shape or form. I imagine integrating a new code editor with our codebase would also be challenging.

here is what I am suggesting: the code is there in the history, let's go ahead with Monaco we can swap the code editor as part of [the bounty issue](https://github.com/asyncapi/studio/issues/684).

TLDR; migration is already complicated enough let's switch to CodeMirror after we migrate to NEXT.

**Comment by KhudaDad414** [comment]
@Amzani it show up as expected now.

**Comment by smoya** [comment]
At a quick glance, I found some differences between versions.

The height of the app is not adjusted in the deploy preview. The Settings button is hidden unless you scroll down (at least in my native MacBook resolution). Same happens with the "errors" and "warnings" bar.

Current production version (https://studio.asyncapi.com):
Google Chrome_Ni6BwwDx@2x

This new version (https://deploy-preview-1062--studio-next.netlify.app/):

Google Chrome_qlE6v65C@2x

The tooltips on the navigation bar differ from one version an the other:

Current production version (https://studio.asyncapi.com):
Google Chrome_LeCqj1vz@2x

This new version (https://deploy-preview-1062--studio-next.netlify.app/):
Google Chrome_PDZuR78z@2x


Additionally, font size or space differs from one version to the other:

Google Chrome_y8OM9V1b

**Comment by KhudaDad414** [comment]
@smoya thanks for the review. 🙇  I can replicate the issues. will make the appropriate changes.

**Comment by smoya** [comment]
> @smoya thanks for the review. 🙇  I can replicate the issues. will make the appropriate changes. 

Nooice! BTW didnt test all functionalities, but just checked the appeal of the page. Meaning it could have more issues (or not).

**Comment by KhudaDad414** [comment]
@smoya done. please take a look when you have the time.

**Review by Amzani — APPROVED** [review]
LGTM
